.travis.yml

```diff
@@ -1,30 +1,111 @@
-dist: xenial
-sudo: true
 language: python
+dist: focal
 
-python:
-  - "3.6"
-  - "3.7"
+stages:
+  - lint
+  - unittest
 
-addons:
-  apt:
-    packages:
-      - libsnappy-dev
-
-install:
-  - pip install flake8 pylint pytest pytest-cov yapf
-  - pip install -r requirements.txt
-
-before_script:
-  - wget https://dev.mysql.com/get/mysql-apt-config_0.8.13-1_all.deb
-  - wget https://repo.percona.com/apt/percona-release_latest.$(lsb_release -sc)_all.deb
+inc_before_install: &inc_before_install
+  # Setup the MySQL repositories
+  - wget -c 'https://repo.mysql.com/mysql-apt-config_0.8.13-1_all.deb'
+  - echo "mysql-apt-config mysql-apt-config/select-server select $MYSQL_VERSION" | sudo debconf-set-selections
+  - echo "mysql-apt-config mysql-apt-config/select-tools select Enabled" | sudo debconf-set-selections
+  - echo "mysql-apt-config mysql-apt-config/select-product select Ok" | sudo debconf-set-selections
+  - echo "mysql-apt-config mysql-apt-config/tools-component string mysql-tools" | sudo debconf-set-selections
   - sudo dpkg -i mysql-apt-config_0.8.13-1_all.deb
+  # Percona tools
+  - wget https://repo.percona.com/apt/percona-release_latest.$(lsb_release -sc)_all.deb
   - sudo dpkg -i percona-release_latest.$(lsb_release -sc)_all.deb
   - sudo percona-release enable-only tools
-  - sudo apt-get update -q
-  - sudo apt-get install -q -y --allow-unauthenticated -o Dpkg::Options::=--force-confnew mysql-server
-  - sudo apt-get install percona-xtrabackup-80
+  - sudo apt update
+
+
+inc_install: &inc_install
+  # Build Deps
+  - sudo apt install -y libsnappy-dev
+  - sudo apt-get install -q -y --allow-unauthenticated -o Dpkg::Options::=--force-confnew mysql-server percona-xtrabackup-80
+  # Setup common python dependencies
+  - pip install -r requirements.txt
+  - pip install -r requirements.dev.txt
+  - pip install -e .
+
+inc_unittest: &inc_unittest
+  - make unittest
+
+jobs:
+  allow_failures:
+    - stage: lint
+      python: 3.6
+      env: [MYSQL_VERSION=mysql-5.7]
```

Comment:
Does setting this env value make Travis do something special or why are we setting this? It looks kind of confusing given that MyHoard only supports MySQL 8.0+

Reply:
The included sections install based on this value (see `inc_install`)

Reply:
Ah, right. I presume something there isn't working quite as expected and it always ends up installing MySQL 8.0 anyway. MyHoard is absolutely incompatible with older MySQL versions and will not work with those.

Reply:
Fixed.